src/nvim/window.c

```diff
@@ -6988,3 +6982,29 @@ win_T *lastwin_nofloating(void) {
   }
   return res;
 }
+
+/*
+ * check if there is at least one unanchored floating window
+ * in the tab page tp.
+ */
+bool check_unanchored_floats(tabpage_T *tp) {
+  FOR_ALL_WINDOWS_IN_TAB(wp, tp) {
+    if (wp->w_floating && wp->w_float_config.relative != kFloatRelativeWindow) {
+      return true;
+    }
+  }
+  return false;
+}
+
+/*
+ * check if there is only one nonfloat in the tabpage tp
+ */
+bool check_one_nonfloat_in_tab(tabpage_T *tp) {
+  int w_nonfloats = 0;
+  FOR_ALL_WINDOWS_IN_TAB(wp, tp) {
+    if(!wp->w_floating) {
+      w_nonfloats++;
+    }
+  }
+  return (w_nonfloats == 1 ? true : false);
```

Comment:
`return (w_nonfloats == 1)`;

Alternatively could check `if(w_nonfloats >= 2) { return false; };` just after the increment and then return true here.